libc/exit/exit.c

```diff
@@ -0,0 +1,1236 @@
+		TEST_ASSERT_EQUAL_INT(1234, buf);
+
+		for (i = 1; i < nodeSize; i++) {
+			TEST_ASSERT_EQUAL_INT(sizeof(int), read(fd, &buf, sizeof(int)));
+			TEST_ASSERT_EQUAL_INT(123, buf);
+		}
+
+		close(fd);
+		remove(TEST_EXIT_PATH);
+	}
+#endif
+}
+
+
+TEST_GROUP_RUNNER(unistd_exit)
+{
+	test_common.test_exitPtr = _exit;
+
+	RUN_TEST_CASE(unistd_exit, status_vals);
+	RUN_TEST_CASE(unistd_exit, exit_status_waitpid);
+	RUN_TEST_CASE(unistd_exit, chk_if_exits);
+	RUN_TEST_CASE(unistd_exit, unblock_thread_wait);
+	RUN_TEST_CASE(unistd_exit, unblock_thread_waitpid);
+	RUN_TEST_CASE(unistd_exit, close_streams);
+	RUN_TEST_CASE(unistd_exit, orphaned_child);
+	RUN_TEST_CASE(unistd_exit, new_parent_id);
+	RUN_TEST_CASE(unistd_exit, SIGCHLD_sent);
+	RUN_TEST_CASE(unistd_exit, per_thread_data_destructors);
+	RUN_TEST_CASE(unistd_exit, no_atexit);
+	RUN_TEST_CASE(unistd_exit, no_flush);
+	RUN_TEST_CASE(unistd_exit, no_handler);
+}
+
+
+TEST_GROUP(unistd_Exit);
+/* Changing test group name to unistd_Exit to make error messages more readable */
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, status_vals)
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, exit_status_waitpid);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, chk_if_exits);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, unblock_thread_wait);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, unblock_thread_waitpid);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, close_streams);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, orphaned_child);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, new_parent_id);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, SIGCHLD_sent);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, per_thread_data_destructors);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, no_atexit);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, no_flush);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, no_handler);
+
+
+TEST_GROUP_RUNNER(unistd_Exit)
+{
+	test_common.test_exitPtr = _Exit;
+
+	RUN_TEST_CASE(unistd_Exit, status_vals);
+	RUN_TEST_CASE(unistd_Exit, exit_status_waitpid);
+	RUN_TEST_CASE(unistd_Exit, chk_if_exits);
+	RUN_TEST_CASE(unistd_Exit, unblock_thread_wait);
+	RUN_TEST_CASE(unistd_Exit, unblock_thread_waitpid);
+	RUN_TEST_CASE(unistd_Exit, close_streams);
+	RUN_TEST_CASE(unistd_Exit, orphaned_child);
+	RUN_TEST_CASE(unistd_Exit, new_parent_id);
+	RUN_TEST_CASE(unistd_Exit, SIGCHLD_sent);
+	RUN_TEST_CASE(unistd_Exit, per_thread_data_destructors);
+	RUN_TEST_CASE(unistd_Exit, no_atexit);
+	RUN_TEST_CASE(unistd_Exit, no_flush);
+	RUN_TEST_CASE(unistd_Exit, no_handler);
+}
+
+
+TEST_GROUP_RUNNER(stdlib_exit)
+{
+	RUN_TEST_CASE(stdlib_exit, stream_flush);
+	RUN_TEST_CASE(stdlib_exit, atexit_few_calls);
+	RUN_TEST_CASE(stdlib_exit, atexit_register_inside);
+	RUN_TEST_CASE(stdlib_exit, atexit_two_nodes);
+}
+
+
+void runner(void)
+{
+	RUN_TEST_GROUP(unistd_exit);
+	RUN_TEST_GROUP(unistd_Exit);
+	RUN_TEST_GROUP(stdlib_exit);
+}
+
+
+int main(int argc, char *argv[])
+{
+	const char *tiniPath = "/tini";
```

Comment:
I'd like to move this code with tini to the PR, where you will enable the ignored cases (when docker devel is ready). For now let's merge the version without it.